packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -448,39 +529,7 @@ class Auth {
         if (accessToken && !this.isTokenExpired(accessToken)) {
             return this.data
         }
-        const refreshTokenRegistered = this.get('refresh_token_registered')
-        const refreshTokenGuest = this.get('refresh_token_guest')
-        const refreshToken = refreshTokenRegistered || refreshTokenGuest
-        if (refreshToken) {
-            try {
-                return await this.queueRequest(
-                    () =>
-                        helpers.refreshAccessToken(
-                            this.client,
-                            {
-                                refreshToken,
-                                ...(this.defaultDnt !== undefined && {dnt: this.defaultDnt})
-                            },
-                            {
-                                clientSecret: this.clientSecret
-                            }
-                        ),
-                    !!refreshTokenGuest
-                )
-            } catch (error) {
-                // If the refresh token is invalid, we need to re-login the user
-                if (error instanceof Error && 'response' in error) {
-                    // commerce-sdk-isomorphic throws a `ResponseError`, but doesn't export the class.
-                    // We can't use `instanceof`, so instead we just check for the `response` property
-                    // and assume it is a fetch Response.
-                    const json = await (error['response'] as Response).json()
-                    if (json.message === 'invalid refresh_token') {
-                        // clean up storage and restart the login flow
-                        this.clearStorage()
-                    }
-                }
-            }
-        }
+        await this.refreshAccessToken()
```

Comment:
Previously, the logic for this is 
```
// get refreshToken
// if refreshToken exist, call refreshToken helper to get a new token, return 
// if refreshToken does not exist, login as guest 

```

now with some refactoring, we are moving the refreshToken logic into a new func, the logic readability looks like that it will retrieve refreshToken **and then** loginGuestUser although it does not work that way. 
```
await this.refreshAccessToken()
return return this.loginGuestUser()
```

Reply:
Thank you for pointing this out. I also returned the loginGuestUser within the refreshAccessToken so a DNT update can handle expired refresh tokens too and call loginGuestUser with new DNT preference.